typescript/sdk/src/deploy/types.ts

```diff
@@ -7,13 +7,33 @@ import type {
 } from '@hyperlane-xyz/core';
 import { Address } from '@hyperlane-xyz/utils';
 
+import { deployInterchainAccount } from '../middleware/account/InterchainAccount';
+import { AccountConfig } from '../middleware/account/types';
+import { MultiProvider } from '../providers/MultiProvider';
 import type { ChainName } from '../types';
 
+export type Owner = Address | AccountConfig;
+
 export type OwnableConfig<Keys extends PropertyKey = PropertyKey> = {
-  owner: Address;
+  owner: Owner;
```

Comment:
instead of editing the base `OwnableConfig` I would only modify the `CoreConfig` with `owner: Address | InterchainAccountConfig` to limit the scope of changes here

Reply:
the core deployer should only have to resolve this address once, and then can pass the address to the hook/ism deployers etc

Reply:
as it stands we are doing the same resolution in many different places

Reply:
yaa, that's true.  We can add `Address | AccountConfig` on a case by case basis.

Reply:
and RouterConfig is also ownableconfig

Reply:
I don't know if it's worth it to reverse the ownableConfig changes I have right now

Reply:
okay, in future lets try to avoid redundant async network interaction